File: modules/agent/agents/metis.go
package agents

import (
	"github.com/woyin/OrangeCoding/modules/ai"
	"github.com/woyin/OrangeCoding/modules/core"
	"github.com/woyin/OrangeCoding/modules/tools"
)

// NewMetis creates the wisdom and judgment agent.
// RoleReviewer, tools: [read_file, grep].
func NewMetis(provider ai.AiProvider, registry *tools.ToolRegistry, workDir string) *BaseAgent {
	allowedTools := []string{"read_file", "grep"}
	return newFilteredAgent(provider, registry, workDir, core.RoleReviewer, allowedTools,
		"You are Metis, the wise counselor. You provide wisdom and judgment, evaluating approaches and advising on best practices.")
}

File: packages/agent/src/agents/metis.ts
/**
 * Metis - wisdom and judgment agent.
 * Ported from modules/agent/agents/metis.go.
 */

import type { AiProvider } from "@orangecoding/ai";
import { AgentRole } from "@orangecoding/core";
import { ToolRegistry } from "@orangecoding/tools";
import { BaseAgent } from "./base.js";
import { newFilteredAgent } from "./helpers.js";

export function newMetis(provider: AiProvider, registry: ToolRegistry, workDir: string): BaseAgent {
  return newFilteredAgent(provider, registry, workDir, AgentRole.Reviewer,
    ["read_file", "grep"],
    "You are Metis, the wise counselor. You provide wisdom and judgment, evaluating approaches and advising on best practices.");
}
